### PR TITLE
Issue291 logout

### PR DESCRIPTION
**Pull request by bitfriend**

### CODE REVIEW

File: app/lib/widgets/SideMenu.dart [hunk 1]
```diff
@@ -312,7 +316,14 @@ class _SideDrawerState extends State<SideDrawer> {
         AppLocalizations.of(context)!.logOut,
         style: SideMenuAndProfileTheme.signOutText,
       ),
-      onTap: () {
+      onTap: () async {
+        if (Get.isRegistered<CrossSigning>()) {
```
Comment (gtalha07) [inline app/lib/widgets/SideMenu.dart:320]: What does this have to do with CrossSigning? Am I missing something here?

Comment (bitfriend) [reply app/lib/widgets/SideMenu.dart:320]: We have to remove all event listeners prior to logout.
If not, event listening is kept after logout and we should get error that it will access invalid objects.

**Review by gtalha07 — APPROVED** [review]
LGTM!

File: app/lib/widgets/SideMenu.dart (hunk 1 above)
Comment (bitfriend) [reply app/lib/widgets/SideMenu.dart:320]: Actually, typing listener or receipt listener is in chat list level and they are deleted when chat list disappears.
But cross signing is in global level so its allocation and deallocation should be dealt globally.
So I allocated it in `main.dart` and deallocated it in logout click.
